### PR TITLE
Ddt 1234

### PR DESCRIPTION
**Pull request by FinnAtAsseco**

### CODE REVIEW

File: tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java [hunk 1]
```diff
@@ -1,23 +1,124 @@
...
+    EquipmentEventTypeCode equipmentEventTypeCode,
+
+    @RequestParam(value = "equipmentReference", required = false) @Size(max = 15)
+    String equipmentReference,
+
+    @RequestParam(value = "limit", defaultValue = "100", required = false) @Min(1)
+    int limit,
+
+    @RequestParam(value = "API-Version", required = false)
+    String apiVersion,
```
Comment (pimg) [inline tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java:81]: This should be a header

File: tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java [hunk 2]
```diff
@@ -1,23 +1,124 @@
...
+
+    paginator.setPageHeaders(request, response, cursor, result);
+    return result.content();
+  }
+
+  /**
+   * Splits a comma-separated string and converts each subpart into
+   * an enum.
+   */
+  private <T extends Enum<T>> List<T> toEnumList(String values, Class<T> enumClass) {
```
Comment (pimg) [inline tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java:112]: fine for now, but possible candidate to move to shared kernel?

File: tnt-persistence/src/main/java/org/dcsa/tnt/persistence/entity/EventCache.java [hunk 3]
```diff
@@ -42,6 +42,9 @@ public class EventCache {
   @Column(name = "content", columnDefinition = "jsonb", nullable = false)
   private String content;
 
+  @Column(name = "document_references", columnDefinition = "text")
+  private String documentReferences;
```
Comment (pimg) [inline tnt-persistence/src/main/java/org/dcsa/tnt/persistence/entity/EventCache.java:46]: Taken line 69 from the EventCachingService do I understand correctly the list of document references is saved as key value pairs delimited with | ?

Comment (FinnAtAsseco) [reply tnt-persistence/src/main/java/org/dcsa/tnt/persistence/entity/EventCache.java:46]: Yes

File: tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java (hunk 1 above)
Comment (FinnAtAsseco) [reply tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java:81]: ok, removed from arguments

File: tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java (hunk 2 above)
Comment (FinnAtAsseco) [reply tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java:112]: Yeah I suspect I will move in when implementing event-subscriptions as I might need it there as well

**Review by pimg — APPROVED** [review]

**Pull request merged**